web-app/src/components/Legend/container.js

```diff
@@ -12,152 +13,89 @@ const LegendContainer = Wrapped =>
 
     constructor(props) {
       super(props);
-      const {
-        PENDING,
-        APPROVED,
-        REJECTED,
-        MANDATORY,
-        WFH,
-        SICK,
-        WRT,
-      } = holidayStatus;
       this.state = {
-        employees: [],
         selectedEmployee: {
           employeeId: -1,
         },
-        eventsKeys: [
-          {
-            eventStatusId: 1,
-            key: PENDING,
-            type: 'event',
-            active: false,
-          },
-          {
-            eventStatusId: 2,
-            key: APPROVED,
-            type: 'event',
-            active: false,
-          },
-          {
-            eventStatusId: 3,
-            key: REJECTED,
-            type: 'event',
-            active: false,
-          },
-          {
-            eventStatusId: 4,
-            key: MANDATORY,
-            type: 'event',
-            active: false,
-          },
-          {
-            eventStatusId: 5,
-            key: WFH,
-            type: 'status',
-            active: false,
-          },
-          {
-            eventStatusId: 6,
-            key: WRT,
-            type: 'status',
-            active: false,
-          },
-          {
-            eventStatusId: 7,
-            key: SICK,
-            type: 'status',
-            active: false,
-          },
-        ],
-        selectedEventStatusID: 1,
+        legendKeys: [],
       };
     }
 
-    componentDidUpdate = (_, prevState) => {
-      const { takenHolidays } = this.props;
-      const { employees } = this.state;
-      const prevEmployees = prevState.employees;
-
-      if (employees.length === prevEmployees.length && employees.length === 0) {
-        this.createEmployeesList(takenHolidays);
-      }
+    componentWillMount = () => {
+      this.buildLegendKeys();
     };
 
-    sortEmployeeList = objArray => {
-      return objArray.sort(function(a, b) {
-        if (a.displayName < b.displayName) return -1;
-        if (a.displayName > b.displayName) return 1;
-        return 0;
-      });
-    };
-
-    createEmployeeObject = hol => {
-      let { employeeId, forename, surname } = hol.employee;
-      return {
-        value: employeeId,
-        displayValue: `${forename} ${surname}`,
-      };
-    };
+    buildLegendKeys = () => {
+      const legendKeys = Object.keys(holidayStatus);
+      const keysFormatted = [];
 
-    removeDuplicateEmployees = employees => {
-      return employees.reduce((unique, o) => {
-        if (
-          !unique.some(
-            obj => obj.value === o.value && obj.displayValue === o.displayValue
-          )
-        ) {
-          unique.push(o);
-        }
-        return unique;
-      }, []);
-    };
+      for (let i = 0; i < legendKeys.length; i++) {
+        const legendKey = legendKeys[i];
```

Comment:
this would be easier to read as a foreach
`legendKeys.forEach((legendKey, index) => {...`